### PR TITLE
refactor(protocol-designer): Use new complib tooltip component

### LINKED ISSUE #5386: bug: PD Step creation button tooltips not centered
# Overview

Tooltip placement for step creation buttons seems to offset by a few pixels as you progress to lower buttons

# Current behavior

Screen Shot 2020-04-08 at 10 23 56 AM
Screen Shot 2020-04-08 at 10 24 04 AM
Screen Shot 2020-04-08 at 10 24 23 AM

# Steps to reproduce

Over over each step creation button and observe progressing offset

# Expected behavior

All tooltips should be centered to the right of each step type

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

closes #5386 by refactoring the tooltips from the older deprecated tooltip with Mikes shiny new one! By doing so we fixed a lot of styling issued that arose.

## changelog

- refactor(protocol-designer): Use new complib tooltip component
- refactor(components): Update types in Buttons.js so that it works with refs/new tooltips properly

## review requests

- [ ] Tooltips for step creation buttons are all now centered on the proper button
- [ ] Nothing in App/PD broke with tooltips!

## risk assessment
Low? This is implementing the new tooltip in one spot in PD and should NOT affect older instances anywhere.

### CODE REVIEW

File: components/src/buttons/Button.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import * as React from 'react'
 import cx from 'classnames'
 import omit from 'lodash/omit'
 
-import type { HoverTooltipHandlers } from '../tooltips'
+// import type { HoverTooltipHandlers } from '../tooltips'
```
Comment (IanLondon) [inline components/src/buttons/Button.js:6]: delete?

File: protocol-designer/src/components/StepCreationButton.js [hunk 2]
```diff
@@ -33,77 +40,79 @@ type Props = {|
...
-      'mix',
-      'pause',
-      'magnet',
-      'temperature',
-      'thermocycler',
-    ]
-    const { isStepTypeEnabled } = this.props
+            const buttonProps = {
+              disabled: disabled,
+              onClick: () => onClick,
```
Comment (IanLondon) [inline protocol-designer/src/components/StepCreationButton.js:74]: this `() => ` is causing the onClick to not work, this line should just be `onClick,`, it's already a fn

File: protocol-designer/src/components/StepCreationButton.js [hunk 3]
```diff
@@ -33,77 +40,79 @@ type Props = {|
...
+        hoverTooltipHandlers={targetProps}
+        onClick={onClick}
+        iconName={stepIconsByType[stepType]}
+        className={cx({
+          [styles.step_button_disabled]: disabled,
+        })}
+      >
+        {i18n.t(`application.stepType.${stepType}`, stepType)}
+      </PrimaryButton>
+      <Tooltip key={stepType} {...tooltipProps}>
```
Comment (IanLondon) [inline protocol-designer/src/components/StepCreationButton.js:111]: is this `key` necessary? Not sure what it's doing here

File: protocol-designer/src/components/StepCreationButton.js [hunk 4]
```diff
@@ -33,77 +40,77 @@ type Props = {|
...
+  ]
+  const { isStepTypeEnabled } = props
 
-  handleExpandClick = (e: SyntheticEvent<>) => {
-    this.setState({ expanded: !this.state.expanded })
-  }
+  return (
+    <div
+      className={styles.list_item_button}
+      onMouseLeave={() => setExpanded(false)}
```
Comment (mcous) [inline protocol-designer/src/components/StepCreationButton.js:60]: This logic behaves weirdly if the z-index of in `styles.list_item_button` above is removed

**Pull request merged**